CruCentralCoast/Community/CommunityGroups/CommunityGroupDetailsVC.swift

```diff
@@ -7,41 +7,87 @@
 //
 
 import UIKit
+import MessageUI
 
-class CommunityGroupDetailsVC: UIViewController {
+class CommunityGroupDetailsVC: UIViewController, MFMessageComposeViewControllerDelegate {
+
+    
 
     @IBOutlet weak var bannerImageView: UIImageView!
-    @IBOutlet weak var genderLabel: UILabel!
-    @IBOutlet weak var dayLabel: UILabel!
-    @IBOutlet weak var timeLabel: UILabel!
+    @IBOutlet weak var dayTimeLabel: UILabel!
     @IBOutlet weak var nameLabel: UILabel!
-    @IBOutlet weak var yearLabel: UILabel!
+    @IBOutlet weak var yearGenderLabel: UILabel!
     @IBOutlet weak var movementLabel: UILabel!
-    @IBOutlet weak var leaderNamesLabel: UILabel!
-    @IBOutlet weak var summaryLabel: UILabel!
+    @IBOutlet weak var joinCommunityGroupButton: CruButton!
     @IBOutlet weak var imageViewAspectRatioConstraint: NSLayoutConstraint!
-
+    
+    var leaderPhoneNumbers : [String] = []
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
         self.navigationItem.largeTitleDisplayMode = .never
+        
     }
     
-    @IBAction func joinCommunityGroup() {
-        // TODO
-        self.presentAlert(title: "Join Community Group", message: "Coming Soon...")
+    @IBAction func didTapContactLeader() {
+        if MFMessageComposeViewController.canSendText() {
+            if self.leaderPhoneNumbers.isEmpty {
+                self.presentAlert(title: "Can't Contact Leader", message: "Sorry, there is no phone number listed for this group")
+            }
+            
+            else {
+                let controller = MFMessageComposeViewController()
+                controller.body = "Hey I'm interested in joining your community group!"
+                controller.recipients = self.leaderPhoneNumbers
+                controller.messageComposeDelegate = self
+                self.present(controller, animated: true, completion: nil)
+            }
+            
+        }
+        else {
+            print("error cant send text")
+        }
+    }
+    
+    func messageComposeViewController(_ controller: MFMessageComposeViewController, didFinishWith result: MessageComposeResult) {
+        controller.dismiss(animated: true, completion: nil)
     }
     
+    
     func configure(with communityGroup: CommunityGroup) {
         DispatchQueue.main.async {
-            self.genderLabel.text = communityGroup.gender.rawValue.uppercased()
-            self.dayLabel.text = communityGroup.weekDay.rawValue.uppercased()
-            self.timeLabel.text = communityGroup.time
-            self.nameLabel.text = communityGroup.name
-            self.yearLabel.text = communityGroup.year.rawValue.uppercased()
+            
+            var leaderArray : [String] = []
+            for leader in communityGroup.leaders {
```

Comment:
I think you could reduce these lines by using flatmap() or compactmap() or map(). Something along the lines of `communityGroup.leaders.compactMap({ $0.name }).joined(separator: ", ")`

Reply:
thanks, good call